### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by michaelmon1203**
Finally got my homework onto Github through Replit!

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -472,9 +338,129 @@ After you have MOVED your 10 UDTs, send me a DM with your pull request link.
...
+    2) number of rehersal time per week
+    3) number of live show time per week
+    4) amount of money made per live show
+    5) amount of money spent for rehersal in studio
+3 things it can do:
+    1) schedule the tour plan
+    2) do live show
+    3) make money
+
+Thing 2) Do YouTube
```
Comment (matkatmusic) [inline main.cpp]: Do Youtube is an activity (action).  
it is not a noun.
- [x] pick a noun

File: main.cpp [hunk 2]
```diff
@@ -472,9 +338,129 @@ After you have MOVED your 10 UDTs, send me a DM with your pull request link.
...
+    2) hour amount on shooting
+    3) hour amount on editing
+    4) view amount per video
+    5) money amount made from per 1000 view
+3 things it can do:
+    1) make video
+    2) increase follower
+    3) make money
 
+Thing 3) Song Writing
```
Comment (matkatmusic) [inline main.cpp]: Song Writing is an activity (action).  
it is not a noun.
- [x] pick a noun

File: main.cpp [hunk 3]
```diff
@@ -472,9 +338,129 @@ After you have MOVED your 10 UDTs, send me a DM with your pull request link.
...
+    2) number of instruments used while producing
+    3) number of BPM
+    4) second amount of the song
+    5) number of people who listened
+3 things it can do:
+    1) reach to the audience
+    2) increase follower
+    3) nail record label's contract
 
+Thing 4) Product Management
```
Comment (matkatmusic) [inline main.cpp]: This is not an ideal object.  
it is too abstract. 
Your properties and actions are describing a **Product Manager.**  
- [x] change the Thing

**Review by matkatmusic — CHANGES_REQUESTED** [review]

**Review by matkatmusic — CHANGES_REQUESTED** [review]
I have added the next set of instructions. 
CODE THEM UP!!

File: main.cpp [hunk 4]
```diff
@@ -1,478 +1,571 @@
...
+    1) reach to the audience
+    2) increase follower
+    3) nail record label's contract
 */
+struct SongWriter
+{
+    int numberOfWordsInLyrics = 300;
+    int numberOfInstrumentsUsedWhileProducing = 6;
+    int secondAmountOfTheSong = 150;
+    int numberOfPeopleWhoListened = 5000;
```
Comment (matkatmusic) [inline main.cpp]: You are missing a property.
and all of your properties are `int`
- [x] revise

File: main.cpp [hunk 5]
```diff
@@ -1,478 +1,571 @@
...
+    int numberOfLiveShowTimePerWeek = 1;
+    int amountOfMoneyMadePerLiveShow = 300;
+    int amountOfMoneySpentForRehersalInStudio = 50;
+
+    void scheduleTheTourPlan(bool ifTheShowIsCancelled = true);
+    void doLiveShow(int showTimeAmount = 120, int songNumbers = 25);
+
+    //returns how much money the band is making
+    float makeMoney(bool isThisShowHasTicket = true);
+};
```
Comment (matkatmusic) [inline main.cpp]: Add a nested type to this UDT
`TourManager` is a good choice
- [x] revise

File: main.cpp [hunk 6]
```diff
@@ -1,478 +1,571 @@
...
+    3) nail record label's contract
 */
+struct SongWriter
+{
+    int numberOfWordsInLyrics = 300;
+    int numberOfInstrumentsUsedWhileProducing = 6;
+    int secondAmountOfTheSong = 150;
+    int numberOfPeopleWhoListened = 5000;
+
+    void reachToTheAudience(int platformAmount, YouTuber recordLabelA);
```
Comment (matkatmusic) [inline main.cpp]: YouTubers aren't record labels.  They're content creators.
Pick a variable name that makes sense.
- [ ] revise

File: main.cpp [hunk 7]
```diff
@@ -1,478 +1,571 @@
...
+    std::string coatType = "ultra-thin";
+
+    struct StringType
+    {
+        bool firstStringWound = false;
+        bool secondStringWound = false;
+        bool thirdStringWound = true;
+        bool fourthStringWound = true;
+        bool fifthStringWound = true;
+        bool sixthStringWound = true;
```
Comment (matkatmusic) [inline main.cpp]: These are not properties relevant to a string type. 
These are properties of a guitar. 
I would remove this nested class entirely, and choose a different UDT to add a nested class to.
- [x] revise

File: main.cpp [hunk 8]
```diff
@@ -1,478 +1,571 @@
...
+        std::string coverColor = "ivory";
+        int stringNumber = 6;
+
+        void pickupStringVibration(bool pluckString = false);
+        void changeTheGuitarTone(int pickupSwitchPosition = 5);
+
+        //returns if pickup cover color matches the body color
+        bool matchTheBodyColor(SongWriter rockWriterA);
+    };
+
```
Comment (matkatmusic) [inline main.cpp]: Delete all of this.
You already declared these classes earlier in the source file. 
you declare instances of THOSE classes.  
You don't copy those class definitions into this class.
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 7 above)
Comment (michaelmon1203) [reply main.cpp]: Ok, I deleted these and added nested class BandManager in RockBand.

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**